### ISSUE TITLE
CUI not present in the owl file

### ISSUE THREAD

**Issue by zoevas** [IHTSDO/snomed-owl-toolkit#75]
I have ran the command for transforming the RDF2 to OWL:

> java -Xms4g -jar snomed-owl-toolkit-3.0.6-executable.jar -rf2-to-owl -rf2-snapshot-archives SnomedCT_RF2Release_INT_20160731.zip

I have tried both the latest version and 2016 version of SNOMED. I noticed that some fields that are present in the online bioportal snomed ontology, are not present in the produced owl.
I am interested for the CUI field so as to link with other ontologies, but is not present in the owl file.
For example, the _cat allergy vaccine_ has the CUI field:
![snomed1](https://user-images.githubusercontent.com/6237917/216848538-1946939f-3d38-499b-be38-ffa2c8148369.PNG)

The produced owl has not the CUI field:
![snomed2](https://user-images.githubusercontent.com/6237917/216848588-fd7ba202-42db-4f26-9f8a-9eef6ec0176a.PNG)

Should I add any other flag when I run the _snomed-owl-toolkit_ in order to have the CUI field?

**Comment by kaicode** [comment]
Hi @zoevas, thanks for reaching out.

The only official SNOMED CT identifier for the concept "Cat allergy vaccine (product)" is `346364006`, or as a URI `http://snomed.info/id/346364006`.

I think the [CUI is a UMLS Concept Unique Identifier](https://www.bioontology.org/wiki/BioPortal_Mappings#Mapping_Sources) that comes from the United States National Library of Medicine. This is not an official SNOMED CT identifier. These are not included in the SNOMED CT release files so can not be populated in the ontology file created by this tool.

I am not aware of a map from the SNOMED CT identifiers to the UMLS CUI identifiers. I've just checked the United States Edition of SNOMED CT and the UMLS website briefly but couldn't find a map. 

I suggest contacting the NLM to see if they can provide guidance.

**Comment by zoevas** [comment]
Hello @kaicode ,

Thank you very mych for your answer. The online bioportal snomed-ct ontology browsing has CUI in all the SNOMED-CT classes. Also, there are additional fields such as _has active ingredient_ etc.
![216848538-1946939f-3d38-499b-be38-ffa2c8148369](https://user-images.githubusercontent.com/6237917/216948162-ab0f9970-bce3-4550-af0f-42221b5f896e.png)


 In case you know, I suppose that this ontology has been created by the NLM, so they might have preprocessed the SNOMED-CT RF and added this CUI field?

Thanks

**Comment by kaicode** [comment]
@zoevas yes, whoever created the ontology in your screenshots must have added the CUI field.
The active ingredient information is part of the definition of this concept but it looks like it's been pulled out into a separate field in this ontology.

**Comment by rwynne** [comment]
@zoevas This view in BioPortal is from an individual source extraction of SNOMED from the UMLS, hence UMLS CUI annotations are present. The BioPortal administrators can tell you more about this process. IIRC, the umls2rdf project (https://github.com/ncbo/umls2rdf) is involved in its creation.

**Comment by zoevas** [comment]
Thank you! I ll ask in this umls2rdf repository how I can incorporate UMLS CUI in SNOMED owl.
